### PR TITLE
Change default FeedbackSize for TripleDES internal implementation

### LINKED ISSUE #43234: CipherMode.CFB does not Encrypt the same in .NET 5.0 RC1 as in .NET 4.7
When using CipherMode.CFB as Mode in the System.Security.Cryptography.TripleDES to Encrypt e.g. a password you don't get the same encryption string back in .NET 5.0 RC1 as you get in .NET 4.7.
We have used code like this since .net 4.5 and it's worked exactly the same since, i should work exactly the same in .NET 5.0 as in 4.x or all apps using this for e.g. saving password encrypted will stop working after upgrade to 
.NET 5.

I did a test console app to reproduce this, it's the exact the same code for both .NET 5.0 RC1 as for .NET 4.7.
Both apps where created in VS 2019 Preview (16..8.0 Preview 3.2).
And they do not generate the same result.
```C#
using System;
using System.Security.Cryptography;
using System.Text;

namespace TestConsole.net5
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello, Typ in you text to be Encrypted!");
            string text = Console.ReadLine();
            TripleDESHelper tripleDESHelper = new TripleDESHelper();
            string encryptedtext = tripleDESHelper.Encrypt(text);
            Console.WriteLine("Your encrypted text results in this: " + encryptedtext);
            Console.ReadLine();
        }
    }
    public class TripleDESHelper
    {

        private readonly byte[] KEY_BYTES = { 2, 24, 1, 114, 15, 74, 23, 43, 81, 23, 87, 34, 55, 72, 32, 34 };
        private readonly byte[] IV_BYTES = { 90, 12, 54, 27, 62, 3, 45, 23 };

        private TripleDES tripleDES;

        public TripleDESHelper()
        {
            tripleDES = TripleDESCryptoServiceProvider.Create();
            tripleDES.Key = KEY_BYTES;
            tripleDES.IV = IV_BYTES;

            // Used for small chunck of data. In this one, we use it for user's password.
            tripleDES.Mode = CipherMode.CFB;
        }

        /// <summary>
        /// Encrypts the given plain text string into an encrypted, base 64 encoded value; using the Triple Data Encryption Standard algorithms
        /// </summary>
        /// <param name="clearText">The plain text.</param>
        /// <returns>A string representing the encrypted text.</returns>
        public string Encrypt(string clearText)
        {
            if (tripleDES == null || String.IsNullOrEmpty(clearText))
                return String.Empty;

            String encryptedText = String.Empty;

            ICryptoTransform encryptor = tripleDES.CreateEncryptor();


            byte[] textBytes = Encoding.UTF8.GetBytes(clearText);

            encryptedText = System.Convert.ToBase64String(encryptor.TransformFinalBlock(textBytes, 0, textBytes.Length));

            return encryptedText;
        }

        /// <summary>
        /// Decrypts the given cipher text string into an plain text using the Triple Data Encryption Standard algorithms
        /// </summary>
        /// <param name="cipherText">The cipher text.</param>
        /// <returns>A string representing the plain text</returns>
        public string Decrypt(string cipherText)
        {
            if (tripleDES == null || String.IsNullOrEmpty(cipherText))
                return String.Empty;

            String decryptedText = String.Empty;

            byte[] encryptedBytes = System.Convert.FromBase64String(cipherText);

            ICryptoTransform decryptor = tripleDES.CreateDecryptor();

            decryptedText = Encoding.UTF8.GetString(decryptor.TransformFinalBlock(encryptedBytes, 0, encryptedBytes.Length));

            return decryptedText;

        }
    }
}
```

### PR DESCRIPTION
**Pull request by vcsjones**
In .NET Framework, TripleDESCryptoServiceProvider defaults to a feedback
size of 8, and TripleDESCng defaults to a feedback size of 64. The
static Create by default would return TripleDESCryptoServiceProvider,
thus the TripleDES from Create would have a default feedback size of 8.

This changes the default sizes of TripleDES to behave more
similarly to .NET Framework to make porting CFB code from Framework
easier.

The internal 3DES implementation (and thus
TripleDESCryptoServiceProvider, since that is a wrapper around the
internal implementation) now defaults to a feedback size of 8.
TripleDESCng and user-derived classes from TripleDES will continue to
use a feedback size of 64.

------

Opening as a draft until tactics can approve for .NET 5. If we can't get this in 5, then we may not want it in 6, either.

Closes #43234

### CODE REVIEW

**Comment by danmosemsft** [comment]
@bartonjs are you planning to write up the template for this, or should someone else ? I think it's fine to stick it here without merging into master, if we'd otherwise potentially revert it. And then just email it.

**Comment by vcsjones** [comment]
@danmosemsft not sure where this should go since this PR is against `master`, but here is my attempt:

## Customer Impact

Reported by a customer in #43234. 

A new mode of symmetric encryption was added to .NET 5, CFB mode. This mode already exists in .NET Framework. When it was brought to .NET 5, the default `FeedbackSize` for some implementations of `TripleDES` differed from the .NET Framework implementation. Customers that were relying on the `FeedbackSize`'s default value in .NET Framework may get incompatible encryption results when porting their .NET Framework code to .NET 5. This is difficult to diagnose - they observed behavior is different results when encrypting data in this mode.

## Testing

The default values for `FeedbackSize` of various algorithms was manually validated against .NET Framework. Unit tests are introduced to solidify the behavior and prevent it from regressing. 

## Risk

Minimal. The default `FeedbackSize` for `TripleDESInternalImplementation` was changed from 64 to 8. Since this mode is new in .NET 5, it does not break compatibility with previous versions of .NET Core. It change the default behavior from previous previews and RCs of .NET 5.

**Comment by vcsjones** [comment]
@bartonjs for reference, here are the feedback sizes from .NET Framework:

```
new DESCryptoServiceProvider().FeedbackSize: 8
new TripleDESCryptoServiceProvider().FeedbackSize: 8
new TripleDESCng().FeedbackSize: 64
new UserDefinedTripleDES().FeedbackSize: 64
new AesCryptoServiceProvider().FeedbackSize: 8
new AesCng().FeedbackSize: 8
new UserDefinedAes().FeedbackSize: 8
```

File: src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/TripleDES/TripleDESCipherTests.cs [hunk 1]
```diff
@@ -827,5 +827,27 @@ public static void MultipleBlockDecryptTransform(bool blockAlignedOutput)
             string decrypted = Encoding.ASCII.GetString(outputBytes, 0, outputOffset);
             Assert.Equal(ExpectedOutput, decrypted);
         }
+
+        [Fact]
+        public static void VerifyNetFxCompat_CFB8_PKCS7Padding()
+        {
+            // .NET Framework would always pad to the nearest block
+            // with CFB8 and PKCS7 padding even though the shorting possible
```
Comment (bartonjs) [inline src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/TripleDES/TripleDESCipherTests.cs:835]:
```suggestion
            // with CFB8 and PKCS7 padding even though the shortest possible
```

**Comment by danmosemsft** [comment]
@bartonjs I'm assuming you'll send the mail for this one, I'm fine with it based on the template. The only other thing to mention explicitly is that fixing it now avoids potentially being forced into a breaking change in 6.0 because of more customer feedback.

**Comment by danmosemsft** [comment]
Approved. @vcsjones can you please de-draftify this PR so we can merge it?

**Comment by vcsjones** [comment]
@danmosemsft done

**Review by bartonjs — APPROVED** [review]

**Pull request merged**